extensions/extension_mgr.go

```diff
@@ -20,45 +22,72 @@ type AdvancedExtension = extensions.AdvancedExtension
 
 const SubstraitDefaultURIPrefix = "https://github.com/substrait-io/substrait/blob/main/extensions/"
 
-// DefaultCollection is loaded with the default Substrait extension
-// definitions. Not all files are currently parsable.
-// Parser needs to enhanced to support all files
-var DefaultCollection Collection
+var (
+	defaultCollection   Collection
+	collectionOnce      sync.Once
+	collectionLoadError error
+)
+
+// GetDefaultCollection returns a Collection that is loaded with the default Substrait extension definitions.
```

Comment:
Could we add a suggestion that folks start using the signature that returns an error ("this version is provided for the ease of use of legacy code")?